[BUG] -validate passes templates that fail to compile at runtime

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Current Behavior

`-validate` reports success for a template that **cannot run**, and exits `0`.

The runtime failure even tells users to run `-validate` to investigate — but `-validate` says the template is fine.

```yaml
# bad.yaml
id: probe-neg
info:
  name: probe
  author: p
  severity: info
http:
  - method: GET
    path: ["{{BaseURL}}"]
    extractors:
      - type: regex
        group: -1
        regex: ["(a)(b)"]
```

Running it:

```
$ nuclei -t bad.yaml -u example.com
[WRN] Could not parse template bad.yaml: could not compile request: could not compile operators:
      could not compile extractor: regex extractor group must be >= 0, got -1
[WRN] Found 1 templates with runtime error (use -validate flag for further examination)
[FTL] Could not run nuclei: no templates provided for scan
```

Validating the same file:

```
$ nuclei -t bad.yaml -validate
[INF] All templates validated successfully
$ echo $?
0
```

### Expected Behavior

`-validate` should fail for a template that cannot be compiled, so template authors and CI catch it before it silently drops out of a scan.

For contrast, `-validate` **does** catch a bad matcher *type*:

```
$ nuclei -t syntaxbad.yaml -validate
[ERR] Error occurred loading template syntaxbad.yaml: Invalid matcher type: nonexistent-type
[FTL] Could not validate templates: errors occurred during template validation
```

So the gap is specifically **value-level errors surfaced during operator compilation**, not structural/type errors.

### Cause

`-validate` goes through `Store.ValidateTemplates` → `Parser.LoadTemplate` → `Parser.ParseTemplate`, which unmarshals the YAML and runs `validateTemplateMandatoryFields`. It never reaches operator compilation.

The compile error is produced later, in `templates.Parse` → `compileProtocolRequests` → `Executer.Compile()`, and on the run path it is counted via `stats.Increment(templates.TemplateRuntimeWarningStats)` in `Store.LoadTemplates` (`pkg/catalog/loader/loader.go:883`). Nothing equivalent happens on the validate path, so `runner.go:713` sees all three stat counters at zero and prints "All templates validated successfully".

Any extractor/matcher error raised at compile time is affected — `group: -1` is just the easiest one to demonstrate, since `CompileExtractors` explicitly rejects it.

### Steps To Reproduce

1. Save the YAML above as `bad.yaml`
2. `nuclei -t bad.yaml -validate` → "All templates validated successfully", exit `0`
3. `nuclei -t bad.yaml -u example.com` → template fails to compile and is dropped

### Environment

```markdown
- Nuclei: dev @ 6ecdb94
- Go: 1.26.4
- OS: darwin/arm64
```

### Anything else?

Impact is mainly on template authoring/CI: `-validate` is the documented gate, and a template can pass it while being unable to run.

Thanks for testing it. The difference is the build, not the behaviour — **`v3.11.0` predates the check this issue is about.**

```
v3.11.0 released    2026-07-06
#7531 merged        2026-07-16   <- added the group >= 0 validation
```

The tag confirms it: `pkg/operators/extractors/compile.go` at `v3.11.0` has no `group must be >= 0` string at all. So on that binary the template is not rejected, which is why your run shows `Templates loaded for current scan: 1` rather than the `[WRN] Could not parse template` line from the report.

## Reproduced on `dev`

Built from current `dev` (`9d9012c`), same file:

```console
$ nuclei -t /tmp/bad.yaml -validate
[INF] All templates validated successfully          # exit 0

$ nuclei -t /tmp/bad.yaml -u example.com
[WRN] Found 1 templates with runtime error (use -validate flag for further examination)
[FTL] Could not run nuclei: no templates provided for scan
```

Same template, same build: `-validate` passes it, the scan refuses it. That gap is the issue.

Side by side on the two builds:

```
v3.11.0   Templates loaded for current scan: 1      <- no validation exists yet
dev       Found 1 templates with runtime error      <- #7531's check fires
```

## One thing worth knowing about v3.11.0 specifically

`group: -1` on that build is not benign — it just did not run in your test, because the template has no matcher, so the extractor is never reached. Add one and it panics:

```console
$ nuclei -t /tmp/probe2.yaml -u https://example.com     # v3.11.0, with a status matcher
panic: runtime error: index out of range [-1]

extractors.(*Extractor).ExtractRegex
    pkg/operators/extractors/extract.go:36
pkg/protocols/http/operators.go:71
pkg/operators/operators.go:253
```

That is what #7531 fixed, and it is why the compile-time check exists. This issue is the follow-on: the check runs during operator compilation, which `-validate` never reaches.

## Scope

To be precise about what is and is not claimed here: `-validate` **does** catch structural errors (a bad matcher `type` fails it correctly). The gap is only value-level errors surfaced during operator compilation — `group: -1` is one instance, and any future check added in `compile.go` would be invisible to `-validate` the same way.

#7603 fixes it by pointing validation at the compiled cache rather than the parsed one. Happy to rebase or adjust it if you would rather take a different approach.

--- a/pkg/catalog/loader/loader.go
+++ b/pkg/catalog/loader/loader.go
@@ -656,7 +656,6 @@ func (store *Store) areTemplatesValid(filteredTemplatePaths map[string]struct{})
 
 func (store *Store) areWorkflowOrTemplatesValid(filteredTemplatePaths map[string]struct{}, isWorkflow bool, load func(templatePath string, tagFilter *templates.TagFilter) (bool, error)) bool {
 	areTemplatesValid := true
-	parsedCache := store.parserCacheOnce()
 
 	for templatePath := range filteredTemplatePaths {
 		if _, err := load(templatePath, store.tagFilter); err != nil {
@@ -666,22 +665,14 @@ func (store *Store) areWorkflowOrTemplatesValid(filteredTemplatePaths map[string
 			}
 		}
 
-		var template *templates.Template
-		var err error
-
-		if parsedCache != nil {
-			if cachedTemplate, _, cacheErr := parsedCache.Has(templatePath); cacheErr == nil && cachedTemplate != nil {
-				template = cachedTemplate
-			}
-		}
-
-		if template == nil {
-			template, err = templates.Parse(templatePath, store.preprocessor, store.config.ExecutorOptions)
-			if err != nil {
-				if isParsingError(store, "Error occurred parsing template %s: %s\n", templatePath, err) {
-					areTemplatesValid = false
-					continue
-				}
+		// The load step validates the parsed definition and filters templates.
+		// FYI parse must still run because protocol compilation can surface
+		// additional validation errors.
+		template, err := templates.Parse(templatePath, store.preprocessor, store.config.ExecutorOptions)
+		if err != nil {
+			if isParsingError(store, "Error occurred parsing template %s: %s\n", templatePath, err) {
+				areTemplatesValid = false
+				continue
 			}
 		}
 

--- a/pkg/catalog/loader/loader_test.go
+++ b/pkg/catalog/loader/loader_test.go
@@ -143,6 +143,60 @@ func TestLoadTemplatesOnlyMetadataLogsCachedTemplateParseErrors(t *testing.T) {
 	require.Contains(t, output.String(), "Could not load template")
 }
 
+func TestValidateTemplatesRejectsRuntimeCompilationError(t *testing.T) {
+	templatePath := filepath.Join(t.TempDir(), "invalid-regex-group.yaml")
+	require.NoError(t, os.WriteFile(templatePath, []byte(`id: invalid-regex-group
+
+info:
+  name: Invalid Regex Group
+  author: pdteam
+  severity: info
+
+http:
+  - method: GET
+    path:
+      - "{{BaseURL}}"
+    extractors:
+      - type: regex
+        group: -1
+        regex:
+          - "(a)(b)"
+`), 0o600))
+
+	options := testutils.DefaultOptions.Copy()
+	var output bytes.Buffer
+	logger := &gologger.Logger{}
+	logger.SetFormatter(formatter.NewCLI(false))
+	logger.SetWriter(&utils.CaptureWriter{Buffer: &output})
+	logger.SetMaxLevel(levels.LevelDebug)
+	options.Logger = logger
+	options.ExecutionId = "loader-invalid-regex-group"
+	options.Templates = []string{templatePath}
+	options.Validate = true
+	options.TemplateLoadingConcurrency = 1
+	testutils.Init(options)
+	t.Cleanup(func() {
+		testutils.Cleanup(options)
+	})
+
+	catalog := disk.NewCatalog("")
+	executerOpts := testutils.NewMockExecuterOptions(options, nil)
+	executerOpts.Catalog = catalog
+	parser := templates.NewParser()
+	parser.ShouldValidate = true
+	executerOpts.Parser = parser
+	executerOpts.Logger = options.Logger
+
+	workflowLoader, err := workflow.NewLoader(executerOpts)
+	require.NoError(t, err)
+	executerOpts.WorkflowLoader = workflowLoader
+
+	store, err := New(NewConfig(options, catalog, executerOpts))
+	require.NoError(t, err)
+	require.Error(t, store.ValidateTemplates())
+	require.Contains(t, output.String(), "regex extractor group must be >= 0")
+}
+
 func TestRemoteTemplates(t *testing.T) {
 	catalog := disk.NewCatalog("")
 

